pkg/core/repo_list_test.go

```diff
@@ -0,0 +1,170 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/oneconcern/datamon/pkg/model"
+	"github.com/oneconcern/datamon/pkg/storage"
+	"github.com/oneconcern/datamon/pkg/storage/mockstorage"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+)
+
+type repoFixture struct {
+	name          string
+	wantError     bool
+	expected      model.RepoDescriptors
+	errorContains []string
+}
+
+func repoTestCases() []repoFixture {
+	return []repoFixture{
+		{
+			name: happyPath,
+			expected: model.RepoDescriptors{
+				{
+					Name:        "myRepo1",
+					Description: "test myRepo1",
+					Contributor: model.Contributor{Email: "test@example.com"},
+				},
+			},
+		},
+		{
+			name:     happyWithBatches,
+			expected: expectedRepoBatchFixture,
+		},
+	}
+}
+
+func buildRepoYaml(repo string) string {
```

Comment:
Why no use the model pkg? I am worried about silent failures as this format is not based on the actual deserialization.

Reply:
Good idea. Have some issues with timestamps as yaml etc, though. Testing that right now